drivers/crypto/crypto_mtls_shim.c

```diff
@@ -325,6 +398,25 @@ static int mtls_session_setup(struct device *dev, struct cipher_ctx *ctx,
 			ctx->ops.ccm_crypt_hndlr = mtls_ccm_decrypt_auth;
 		}
 		break;
+	case CRYPTO_CIPHER_MODE_GCM:
+#ifdef CONFIG_MBEDTLS_CIPHER_MODE_GCM_ENABLED
+		gcm_ctx = &mtls_sessions[ctx_idx].mtls_gcm;
+		mbedtls_gcm_init(gcm_ctx);
+		ret = mbedtls_gcm_setkey(gcm_ctx, MBEDTLS_CIPHER_ID_AES,
+					 ctx->key.bit_stream, ctx->keylen * 8U);
+		if (ret) {
+			LOG_ERR("Could not setup the key (%d)", ret);
+			mtls_sessions[ctx_idx].in_use = false;
+
+			return -EINVAL;
+		}
+		if (op_type == CRYPTO_CIPHER_OP_ENCRYPT) {
+			ctx->ops.gcm_crypt_hndlr = mtls_gcm_encrypt_auth;
+		} else {
+			ctx->ops.gcm_crypt_hndlr = mtls_gcm_decrypt_auth;
+		}
+#endif /* CONFIG_MBEDTLS_CIPHER_MODE_GCM_ENABLED */
```

Comment:
This shouldn't be needed, because of the previous check this code will never be reached if `CONFIG_MBEDTLS_CIPHER_MODE_GCM_ENABLED` is undefined.

Reply:
Sorry for being imprecise, `mtls_session_setup()` will return `-EINVAL` when `CONFIG_MBEDTLS_CIPHER_MODE_GCM_ENABLED` is not defined, because of the mode check before the switch statement.

Reply:
Fair enough.